lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1320,17 +1418,22 @@ def _update_relation_data(self, _):
         if not self._charm.unit.is_leader():
             return
 
-        if alert_groups := load_alert_rules_from_dir(
-            self.dir_path, self.topology, recursive=self._recursive
-        ):
+        rules_file = (
+            AlertRules(self.topology)
+            .add_from_path(self.dir_path, recursive=self._recursive)
+            .as_dict()
+        )
+
+        if rules_file:
             logger.info("Updating relation data with rule files from disk")
             for relation in self._charm.model.relations[self._relation_name]:
                 relation.data[self._charm.app]["alert_rules"] = json.dumps(
-                    {"groups": alert_groups},
+                    [self.topology.identifier, rules_file],
```

Comment:
It is not clear why topology information needs to be provided here when it already available in the scrape metadata.

Reply:
That approach was already reverted in the previous commit, but I overlooked this line. Thanks!